### PR TITLE
[terra-application-layout] Add Theme context

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Added theme context  to terra application layout

Closes #

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-application-layout/src/clinical-lowlight-theme/ApplicationLayout.module.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    --terra-application-layout-user-data-photo-width: 3.571rem;
+    --terra-application-layout-user-data-info-margin-right: 0;
+    --terra-application-layout-user-data-name-font-size: 1.143rem;
+    --terra-application-layout-user-data-name-font-weight: normal;
+    --terra-application-layout-user-data-detail-font-size: 0.8571rem;
+
+
+
+
+
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/clinical-lowlight-theme/ApplicationLayout.module.scss:58]: remove empty lines

File: packages/terra-application-layout/src/header/ApplicationHeader.module.scss [hunk 2]
```diff
@@ -1,5 +1,8 @@
 @import '~terra-breakpoints/lib/media-queries';
 
+// Themes
+@import '../clinical-lowlight-theme/ApplicationLayout.module';
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/header/ApplicationHeader.module.scss:4]: the fusion-theme also needs to be applied

File: packages/terra-application-layout/src/header/_ApplicationHeader.jsx [hunk 3]
```diff
@@ -117,13 +119,14 @@ class ApplicationHeader extends React.Component {
...
+    const theme = this.context;
 
     if (layoutConfig.toggleMenu) {
       return (
-        <div className={cx('navbar-toggle')}>
+        <div className={classNames(cx('navbar-toggle', theme.className))}>
           <button
             type="button"
-            className={cx('toggle-button')}
+            className={classNames(cx('toggle-button', theme.className))}
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/header/_ApplicationHeader.jsx:129]: The theme will be applied by the parent div https://github.com/cerner/terra-framework/pull/1116/files#diff-e660d321a39cefac3535a9755c8a3fcfR126
```suggestion
            className={cx('toggle-button')}
```

File: packages/terra-application-layout/src/header/_ApplicationHeader.jsx [hunk 4]
```diff
@@ -117,13 +119,14 @@ class ApplicationHeader extends React.Component {
 
   renderToggle() {
     const { layoutConfig, intl } = this.props;
+    const theme = this.context;
 
     if (layoutConfig.toggleMenu) {
       return (
-        <div className={cx('navbar-toggle')}>
+        <div className={classNames(cx('navbar-toggle', theme.className))}>
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/header/_ApplicationHeader.jsx:126]: The theme will be applied by the parent div https://github.com/cerner/terra-framework/pull/1116/files#diff-e660d321a39cefac3535a9755c8a3fcfR245
```suggestion
        <div className={cx('navbar-toggle')}>
```

File: packages/terra-application-layout/src/menu/ApplicationMenu.module.scss [hunk 5]
```diff
@@ -1,3 +1,6 @@
+// Themes
+@import '../clinical-lowlight-theme/ApplicationLayout.module';
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/menu/ApplicationMenu.module.scss:2]: missing fusion-theme

**Comment by saket2403** [comment]
Resolved all comments ee11d3b

File: packages/terra-application-layout/package.json [hunk 6]
```diff
@@ -42,7 +42,8 @@
     "terra-modal-manager": "^6.33.0",
     "terra-navigation-layout": "^5.23.0",
     "terra-navigation-side-menu": "^2.30.0",
-    "terra-popup": "^6.33.0"
+    "terra-popup": "^6.33.0",
+    "terra-theme-context": "^1.2.0"
```
Comment (mjhenkes) [inline packages/terra-application-layout/package.json:46]: As you're adding this dependency, make sure that all packages are on *exactly* the same version i the lerna repo. There is a quirk with lerna hoisting that won't resolve the minor versions which will result in multiple versions of the context being pulled into the test site, breaking some of the components.

Comment (saket2403) [reply packages/terra-application-layout/package.json]: Fixed 4e8bf43

File: packages/terra-application-layout/src/header/ApplicationHeader.module.scss [hunk 7]
```diff
@@ -1,5 +1,9 @@
...
+@import '../orion-fusion-theme/ApplicationLayout.module';
+
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/header/ApplicationHeader.module.scss:6]:
```suggestion
@import '../clinical-lowlight-theme/ApplicationHeader.module';
@import '../orion-fusion-theme/ApplicationHeader.module';

```

File: packages/terra-application-layout/src/menu/ApplicationMenu.module.scss [hunk 8]
```diff
@@ -1,3 +1,7 @@
...
+@import '../orion-fusion-theme/ApplicationLayout.module';
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/menu/ApplicationMenu.module.scss:3]:
```suggestion
@import '../clinical-lowlight-theme/ApplicationMenu.module';
@import '../orion-fusion-theme/ApplicationMenu.module';
```

File: packages/terra-application-layout/src/user/UserData.module.scss [hunk 9]
```diff
@@ -1,3 +1,7 @@
+// Themes
+@import '../clinical-lowlight-theme/ApplicationLayout.module';
+@import '../orion-fusion-theme/ApplicationLayout.module';
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/user/UserData.module.scss:3]:
```suggestion
@import '../clinical-lowlight-theme/UserData.module';
@import '../orion-fusion-theme/UserData.module';
```

File: packages/terra-application-layout/src/clinical-lowlight-theme/ApplicationLayout.module.scss [hunk 10]
```diff
@@ -0,0 +1,55 @@
+:local {
+  .clinical-lowlight-theme {
```
Comment (emilyrohrbough) [inline packages/terra-application-layout/src/clinical-lowlight-theme/ApplicationLayout.module.scss:2]: Lets create a theme for for each scss file:
```
ApplicationHeader.module.scss
ApplicationMenu.scss
UserData.scss
```

**Comment by saket2403** [comment]
Separated files here 5913545

**Pull request merged**